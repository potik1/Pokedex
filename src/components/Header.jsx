import React from 'react';
import SearchPokemonByName from './SearchPokemonByName';

const Header = () => (
  <nav className="navbar bg-secondary text-white">
    <a className="navbar-brand">POKEDEX</a>
      <SearchPokemonByName  />
  </nav>
);

export default Header;
