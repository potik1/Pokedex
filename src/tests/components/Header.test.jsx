import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

describe('Header', () => {
  const header = shallow(< Header />);

  test('renders correctly', () => {
    expect(header).toMatchSnapshot();
  });

  test('contains Connected SearchPokemonByName component', () => {
    expect(header.find('Connect(SearchPokemonByName)').exists()).toBe(true);
  });
});
