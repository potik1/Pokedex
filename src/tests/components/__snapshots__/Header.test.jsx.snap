// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navbar bg-info "
>
        <a
                className="navbar-brand text-white mx-3"
                href="/"
        >
                POKEDEX
        </a>
        <Connect(SearchPokemonByName) />
</nav>,
      "className": "container px-0",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="navbar-brand text-white mx-3"
            href="/"
>
            POKEDEX
</a>,
          <Connect(SearchPokemonByName) />,
        ],
        "className": "navbar bg-info ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "POKEDEX",
            "className": "navbar-brand text-white mx-3",
            "href": "/",
          },
          "ref": null,
          "rendered": "POKEDEX",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar bg-info "
>
          <a
                    className="navbar-brand text-white mx-3"
                    href="/"
          >
                    POKEDEX
          </a>
          <Connect(SearchPokemonByName) />
</nav>,
        "className": "container px-0",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="navbar-brand text-white mx-3"
              href="/"
>
              POKEDEX
</a>,
            <Connect(SearchPokemonByName) />,
          ],
          "className": "navbar bg-info ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "POKEDEX",
              "className": "navbar-brand text-white mx-3",
              "href": "/",
            },
            "ref": null,
            "rendered": "POKEDEX",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
